src/lib/common/sol-platform-linux-common.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int sol_platform_impl_linux_set_hostname(const char *name);
+
+const char *sol_platform_impl_linux_get_hostname(void);
```

Comment:
no need to add this file, the `set_hostname()` is not in common, the `sol_platform_impl_get_hostname()` (without `_linux`!) is already in the `sol-platform-impl.h`